scripts/behavior.js

```diff
@@ -381,6 +463,123 @@ function executeCommand(command, arguments, cooldown, message, settings, utils,
+            });
+    });
+}
+
+/**
+ * Deletes a role in the specifics and on the discord server
+ * Won't check if the role is present
+ * @param {String} roleName The name of the role to delete
+ * @param {Message} message The message from which to parse guild and author
+ * @returns {Promise<any>} A promise that completes successfully when the role deletion succeeded and completes exceptionally when the deletion failed due to the discord server
+ */
+function deleteRole(roleName, message) {
+    return new Promise((resolve, reject) => {
+        let specifics = utils.getSpecifics(message.guild);
+        specifics.registeredRoles = specifics.registeredRoles.filter(role => role !== roleName);
+        utils.writeSpecifics(message.guild, specifics);
+
+        let roleToDelete = Array.from(message.guild.roles.values()).find(role => role.name === roleName);
+        if(roleToDelete !== undefined) {
+            roleToDelete.delete()
+                .then(v => {
+                    utils.log("Role ${roleName} deleted.", "--", message.guild);
+                    replyToMessage(message, "Role deleted").then(resolve());
+                });
+        } else {
+            utils.log("Role ${roleName} unregistered. Could not delete the discord role. (not found)", "--", message.guild);
+            replyToMessage(message, "Role was unregistered from Dostya. I could not delete the discord role - Maybe permissions are missing? Please remove the discord role manually.")
+                .then(reject());
+        }
+    });
+}
+
+/**
+ * Subscribe to a role (assign discord role)
+ * Won't check if the role is present (If called without checking beforehand this can assign potentially hazardous roles like administrator to users )
+ * @param {String} roleName The name of the role
+ * @param {Message} message The message from which to parse guild and author
+ * @returns {Promise<any>} A promise that completes successfully when the role (un)subscription and completes exceptionally when the (un)subscription failed due to the discord server
+ */
+async function subscribe(roleName, message) {
+    let roleToSubscribe = Array.from(message.guild.roles.values()).find(role => role.name === roleName);
+
+    if(roleToSubscribe !== undefined) {
+        let guildMember = await message.guild.fetchMember(message.author);
+        await guildMember.addRole(roleToSubscribe);
+        utils.log("Added member ${message.author.id} to role ${roleName}.", "--", message.guild);
+        await replyToMessage(message, "You successfully subscribed to role " + roleName);
+        return;
+    } else {
+        utils.log("Failed adding member ${message.author.id} to role ${roleName}. (role not found)", "--", message.guild);
+        await replyToMessage(message, "Role not present on discord server. Please ask a moderator for help. (role is registered with this bot but wasn't found in guild/server roles)");
+        throw "Role not present on discord server.";

```

Comment:
If it has been logged already maybe no need to specify message in the throw

Reply:
it doesn't hurt aswell if the message is catched sometime in the future